spec.md

```diff
@@ -542,6 +542,172 @@ The following string-encoded integer literals all have the value `-42`:
 Radix-specified integer literals are only usable when constructing hardware
 integer literals.  Any use in place of an integer is disallowed.
 
+# Circuit Components
+
+Circuit components are the named parts of a module corresponding to physical hardware.
+
+There are seven **kinds** of circuit components.
+They are: nodes, wires, registers, output ports, input ports, submodule instances, and memories.
+
+Circuit components can be connected together in the hardware (see [@sec:connects]).
+
+Each circuit component in a module is associated
+with a type called its type ([@sec:types]).
+This is the type of data that travels across the component.
+It is used to determine the legality of connections.
+
+
+## Nodes
+
+Nodes are named expressions in FIRRTL.
+
+Example:
+
+```firrtl
+    node mynode = and(in, UInt<4>(1))
+```
+
+The type of a node is the type of the expression given in the definition.
+
+
+## Wires
+
+Wires represent named expressions whose value is determined
+by FIRRTL `connect`{.firrtl} statements (see [@sec:connects]).
+Also unlike nodes, the type of a wire must be explicitly declared.
+
+Example:
+
+``` firrtl
+    wire mywire: UInt<1>
+
+    connect mywire, UInt<1>(0)
+```
+
+The type of a wire is given after the colon (`:`{.firrtl}).
+
+
+## Registers
+
+Registers are stateful elements of a design.
+
+Registers can be connected (see [@sec:connects]).
+The state of the register is controlled through what is connected to it.
+
+The state of a register may be any non-`const`{.firrtl} passive type (see [@sec:passive-types]).
+Registers are always associated with a clock.
+Optionally, registers may have a reset signal.
+
+On every cycle, a register will drive its current value
+and then latch the value it will take on for the next cycle.
+
+The `regreset` keyword is used to declare a register with a reset.
+The `reg` keyword is used to declare a register without a reset.
+
+Examples:
+
+``` firrtl
+wire myclock : Clock
+reg myreg : SInt, myclock
+```
+
+``` firrtl
+wire myclock : Clock
+reg myreg : SInt, myclock
+```
+```firrtl
+wire myclock : Clock
+wire myreset : UInt<1>
+wire myinit : SInt
+regreset myreg : SInt, myclock, myreset, myinit
+```
+
+For both variants of register, the type is given after the colon (`:`{.firrtl}).
+
+Semantically, registers become flip-flops in the design.
+The the next value is latched on the positive edge of the clock.
+For registers without a reset, the initial value is indeterminate (see [@sec:indeterminate-values]).
```

Comment:
There was some very carefully constructed language in the original register statement section that intentionally does not talk about "initial value" and instead uses "reset value". This is extremely intentional as the initial value of a register is assumed to mean "the value of the register before any reset is applied" or "the value of the register at time zero".

Put differently, the initial value of both registers with and without resets is indeterminate.

Reply:
Whoops.